py/runai_model_streamer/runai_model_streamer/file_streamer/file_streamer.py

```
@@ -140,15 +190,25 @@ def get_chunks(self) -> Iterator:
             )
 
+        # Finalize all remaining cache writers after streaming completes.
+        if self._cache.enabled:
+            for path in list(self._cache._writers.keys()):
+                self._cache.finalize(path)
+
     # This function iterates over indexes of ready chunks.
     # The indexes are relative to the last request that sent
     # And need to be translated to global index in the chunks list
     def request_ready_chunks(self) -> Iterator:
         for i in range(sum(len(file_request.chunks) for file_request in self.active_request.files)):
```

Ruff (0.15.18)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)